src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php

```diff
@@ -296,9 +297,25 @@ private function getContainerEnvVars(ContainerBuilder $container): array
             return [];
         }
 
-        $file = file_get_contents($container->getParameter('debug.container.dump'));
-        preg_match_all('{%env\(((?:\w++:)*+\w++)\)%}', $file, $envVars);
-        $envVars = array_unique($envVars[1]);
+        if(!$container->hasParameter('kernel.project_dir') || !$container->hasParameter('kernel.environment')) {
+            return [];
+        }
+
+        $envVarsFromContainer = [];
+        $containerFile = file_get_contents($container->getParameter('debug.container.dump'));
+        preg_match_all('{%env\(((?:\w++:)*+\w++)\)%}', $containerFile, $envVarsFromContainer);
+        $envVarsFromContainer = array_unique($envVarsFromContainer[1]);
+
+        $envFiles = $this->getEnvFiles($container->getParameter('kernel.project_dir'), $container->getParameter('kernel.environment'));
+        $availableFiles = array_filter($envFiles, static fn (string $file) => is_file($file));
+
+        $dotenv = new Dotenv();
+        $envVarsFromFiles = [];
+        foreach ($availableFiles as $file) {
+            $envs = $dotenv->parse(file_get_contents($file), $file);
```

Comment:
You don't need to find and parse all the env files. `SYMFONY_DOTENV_VARS` already contains all the variables read by `Dotenv`.

https://github.com/symfony/symfony/blob/acb7a989976f5f88cf745e9580a4ece23d65e12a/src/Symfony/Component/Dotenv/Command/DebugCommand.php#L152-L158

Note that the `.env` file names can be customized with the [runtime options](https://symfony.com/doc/current/components/runtime.html#using-options) `dotenv_path`.